File: 프로그래머스/1/12982. 예산/예산.java
import java.util.*;

class Solution {
    public int solution(int[] d, int budget) {
        int answer = 0;
        int count = 0;
        Arrays.sort(d);
        
        for(int i = 0; i < d.length; i++){
            if(count + d[i] > budget){
                break;
            }
            count += d[i];
            answer++;
        }
        return answer;
    }
}

File: 프로그래머스/1/12982. 예산/예산.py
# greedy 사용

def solution(d, budget):
    answer = 0
    total = 0
    
    d.sort()
    for i in range(0, len(d)):
        if total + d[i] > budget:
            break
        total += d[i]
        answer += 1
        
    return answer